lib/src/model/broadcast/broadcast_providers.dart

```diff
@@ -2,20 +2,50 @@ import 'package:fast_immutable_collections/fast_immutable_collections.dart';
 import 'package:lichess_mobile/src/model/broadcast/broadcast.dart';
 import 'package:lichess_mobile/src/model/broadcast/broadcast_repository.dart';
 import 'package:lichess_mobile/src/model/common/http.dart';
+import 'package:lichess_mobile/src/model/common/id.dart';
 import 'package:riverpod_annotation/riverpod_annotation.dart';
 
 part 'broadcast_providers.g.dart';
 
 @riverpod
-Future<IList<Broadcast>> broadcasts(BroadcastsRef ref) async {
-  return ref
-      .withClient((client) => BroadcastRepository(client).getBroadcasts());
+class BroadcastPage extends _$BroadcastPage {
+  @override
+  int build() {
+    return 1;
+  }
+
+  void next() {
+    if (state < 20) {
+      // 20 is the last page
+      state = state + 1;
+    }
+  }
+}
+
+@riverpod
+class BroadcastsList extends _$BroadcastsList {
+  @override
+  Future<BroadcastResponse> build() async {
+    final page = ref.watch(broadcastPageProvider);
+
+    final broadcastResponse = await ref.withClient(
+      (client) => BroadcastRepository(client).getBroadcasts(page: page),
+    );
+
+    if (page == 1) {
+      return broadcastResponse;
+    } else {
+      return state.requireValue.copyWith(
```

Comment:
There is a risk that this provider is initialised with a page that is not 1 and then it will throw an exception.

Reply:
I don't think it can happen because I watch the `broadcastListProvider` before the `broadcastPageProvider`.

Do you have an example of how it could happen?

Reply:
Don't know, it just feels dangerous. It may be ok now, but could be not ok in the future, or with some edge case. We have to think about maintenance.

Also it is the first time I see this pattern where the previous state is used in the `build` method, which is not what we normally do.

The way it works is clever but I think it can be simplified. `BroadcastPage` and `BroadcastList` can be merged together. The fewer providers the better. The problem with your solution (on top of what I mentioned earlier) is that it is not obvious that state is accumulated from an `int` provider. And the page provider is a bit confusing since we don't really need to remember the page we're at, but the last downloaded page. It can be part of the `BroadcastList` state, by returning a tuple for instance. And we don't need to use our own logic for the `page` since the server handles it. Use the `currentPage` and `nextPage` fields in the json response.

Reply:
I remember why I split it in two providers. I thought that the current page should not be part of the state because the page number is not going to be used elsewhere than in the provider.

But I agree that it probably makes the code harder to understand. I'll create a `BroadcastListState` that will hold the the page number and the list of tournament broadcast so there will be only one provider.